pages/utilities/regex-tester.tsx

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useState } from "react";
+import { Textarea } from "@/components/ds/TextareaComponent";
+import PageHeader from "@/components/PageHeader";
+import { Card } from "@/components/ds/CardComponent";
+import { Button } from "@/components/ds/ButtonComponent";
+import { Label } from "@/components/ds/LabelComponent";
+import Header from "@/components/Header";
+import { CMDK } from "@/components/CMDK";
+import Meta from "@/components/Meta";
+import { useCopyToClipboard } from "@/components/hooks/useCopyToClipboard";
+
+export default function RegexTester() {
+  const [pattern, setPattern] = useState("");
+  const [testString, setTestString] = useState("");
+  const [result, setResult] = useState<string | null>(null);
+  const { buttonText, handleCopy } = useCopyToClipboard();
+
+  const handlePatternChange = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setPattern(event.currentTarget.value);
+    },
+    []
+  );
+
+  const handleTestStringChange = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setTestString(event.currentTarget.value);
+    },
+    []
+  );
+
+  const handleTest = useCallback(() => {
+    try {
+      const regex = new RegExp(pattern);
+      const match = regex.test(testString);
+      setResult(match ? "Match found" : "No match found");
+    } catch (error) {
+      setResult("Invalid regex pattern");
+    }
+  }, [pattern, testString]);
```

Comment:
Since this function covers very basic use case, my suggestion is to extend this function and allow users to pass full regex patterns with flags. With this approach we will have better error handling / error messaging.

First, let's add function outside of the react component, to the bottom of the file and name it `createRegex()`

```
const createRegex = (pattern: string): RegExp => {
  if (!pattern) {
    throw new Error("Pattern cannot be empty");
  }

  if (pattern.startsWith("/") && pattern.lastIndexOf("/") !== 0) {
    const lastSlashIndex = pattern.lastIndexOf("/");
    const patternBody = pattern.slice(1, lastSlashIndex);
    const flags = pattern.slice(lastSlashIndex + 1);

    if (!/^[gimsuy]*$/.test(flags)) {
      throw new Error("Invalid regex flags");
    }

    return new RegExp(patternBody, flags);
  } else {
    return new RegExp(pattern);
  }
};
```

Then utilize function in `handleTest()`, I think it makes sense to add matches as part of `result`
For better error handling, let's use `error.message` as output instead of generic `Invalid regex pattern`

```
const handleTest = useCallback(() => {
  try {
    const regex = createRegex(pattern);
    const matches = testString.match(regex);

    if (matches) {
      setResult(`Match found:\n${matches.join("\n")}`);
    } else {
      setResult("No match found");
    }
  } catch (error) {
    if (error instanceof Error) {
      setResult(error.message);
    }
  }
}, [pattern, testString]);
```